include/grpc/grpc_security.h

```diff
@@ -834,30 +838,73 @@ typedef void (*grpc_tls_on_credential_reload_done_cb)(
     grpc_tls_credential_reload_arg* arg);
 
 /** A struct containing all information necessary to schedule/cancel a
-    credential reload request.
-    - cb and cb_user_data represent a gRPC-provided
-      callback and an argument passed to it.
-    - key_materials_config is an in/output parameter containing currently
-      used/newly reloaded credentials. If credential reload does not result in
-      a new credential, key_materials_config should not be modified. The same
-      key_materials_config object can be updated if new key materials is
-      available.
-    - status and error_details are used to hold information about
-      errors occurred when a credential reload request is scheduled/cancelled.
-    - config is a pointer to the unique grpc_tls_credential_reload_config
-      instance that this argument corresponds to.
-    - context is a pointer to a wrapped language implementation of this
-      grpc_tls_credential_reload_arg instance.
-    - destroy_context is a pointer to a caller-provided method that cleans
-      up any data associated with the context pointer.
+    credential reload request. To help understand each fields, we mark each
+    field with labels. Here are their explanations
+    -------------------------label explanation----------------------------------
+    [USER-MANAGED]: fields that are managed directly by the end users
+    [SYSTEM-PROVIDED]: fields that needed to retrieve implementations in C core
+    and would be provided by the gRPC stack
+    [WRAP-LANG]: fields that needed to retrieve the particular implementation in
+    wrap language
+    [ASYNC_ONLY]: fields that only used when in asynchronous mode. If a field
+    doesn't have this label, we can assume it should be applied to both
+    synchronous and asynchronous mode.
+    Note that if a field is marked both as [USER-MANAGED] and [SYSTEM-PROVIDED],
+    it indicates that the field is initiated by the gRPC stack, but users
+    need to properly manage them to correctly perform credential reloading.
+    ----------------------label explanation ends--------------------------------
+    - cb[USER-MANAGED][ASYNC_ONLY]: a callback-function provided for users to
+      indicate an asynchronous reloading operation is complete. After calling
+      this function, gRPC stack would be notified and continue rest of the
+      authentication.
+    - cb_user_data[SYSTEM-PROVIDED][ASYNC_ONLY]: a pointer used to retrieve the
+      corresponding security connector implementation. This is needed because
+      we need to use the |peer_checked| closure in security connector after
```

Comment:
`peer_checked`? because this was copy-pasted from the other auth check struct?

Reply:
Nah, peer_checked is the callback function attached to secuity_connector: https://github.com/grpc/grpc/blob/master/src/core/lib/security/security_connector/tls/tls_security_connector.h#L100
The reason why we need this field is, we need to use this  `on_peer_checked` at some point of time, so we keep a pointer to the security_connector, and use that to get `on_peer_checked`.
May I know if there are any suggestions for rephrasing this?

Reply:
The connection between `peer_checked`, security-connector and reload config is still not very clear so I kind of agree with @yihuazhang , that the mention of Security manager and `peer_checked` in the context of reload config is more confusing and if users don't need to know this, should we mention it here?

Reply:
OK, Got you!